src/ripple/nodestore/impl/Database.cpp

```diff
@@ -296,7 +301,9 @@ Database::threadEntry()
     while (true)
     {
         uint256 lastHash;
-        std::uint32_t lastSeq;
+        std::vector< std::pair <std::uint32_t,
+            std::function<void(std::shared_ptr<NodeObject>&)>>> entry;
+
```

Comment:
I believe we can eliminate `lastHash` and move the `entry` declaration outside the loop.
eg.
```
    std::vector<std::pair<
        std::uint32_t,
        std::function<void(std::shared_ptr<NodeObject>&)>>>
        entry;
    while (true)
    {
        {
            std::unique_lock<std::mutex> lock(readLock_);
            while (!readShut_ && read_.empty())
            {
                // All work is done
                readCondVar_.wait(lock);
            }
            if (readShut_)
                break;

            // Read in key order to make the back end more efficient
            auto it = read_.lower_bound(readLastHash_);
            if (it == read_.end())
            {
                // start over from the beginning
                it = read_.begin();
            }
            readLastHash_ = it->first;
            entry = std::move(it->second);
            read_.erase(it);
        }

        for (auto const& req : entry)
        {
            auto obj =
                fetchNodeObject(readLastHash_, req.first, FetchType::async);
```